server/src/main/java/org/apache/druid/catalog/model/TableDefnRegistry.java

```diff
@@ -102,10 +144,25 @@ public ResolvedTable resolve(TableSpec spec)
     if (Strings.isNullOrEmpty(type)) {
       throw new IAE("The table type is required.");
     }
-    TableDefn defn = defns.get(type);
+    TableDefn defn = tableDefns.get(type);
     if (defn == null) {
       throw new IAE("Table type [%s] is not valid.", type);
     }
     return new ResolvedTable(defn, spec, jsonMapper);
   }
+
+  public InputSourceDefn inputSourceDefnFor(String type)
+  {
+    return type == null ? null : inputSourceDefns.get(type);
```

Comment:
nit: I dont think you need the null check do you? If null, should just return null? Or why is this check not being done for formats()?